packages/manager/modules/pci/src/projects/onboarding/onboarding.controller.js

```diff
@@ -1,7 +1,59 @@
 import illustration from './onboarding.png';
 
 export default class {
-  constructor() {
+  /* @ngInject */
+  constructor($q, orderCart, pciProjectNew) {
     this.illustration = illustration;
+    this.$q = $q;
+    this.orderCart = orderCart;
+    this.pciProjectNew = pciProjectNew;
+  }
+
+  onCreateProjectClick() {
+    return this.goToCreateNewProject();
+  }
+
+  // Waiting for specification to clarify the behaviour of the HDS component
+  setHdsOptionItem() {
+    const { cartId, projectItem, hdsItem } = this.cart;
+    const { hds: hdsIsChecked } = this.model;
+    if (this.hds.isValidSupportLevel) {
+      if (hdsIsChecked && !hdsItem) {
+        const priceMode = this.hds.option.prices.find(({ capacities }) =>
+          capacities.includes('renew'),
+        );
+        return this.orderCart
+          .addNewOptionToProject(cartId, {
+            duration: priceMode.duration,
+            itemId: projectItem.itemId, // Add HDS option to project
+            planCode: this.hds.option.planCode,
+            pricingMode: priceMode.pricingMode,
+            quantity: 1,
+          })
+          .then((option) => this.cart.addItem(option));
+      }
+      if (!hdsIsChecked && hdsItem) {
+        return this.orderCart
+          .deleteItem(cartId, hdsItem.itemId)
+          .then(() => this.cart.deleteItem(hdsItem));
+      }
+    }
+    return this.$q.when();
+  }
+
+  onHdsCheckboxChanged(hdsCheckbox) {
+    this.model.hds = hdsCheckbox.status;
+    this.hds.isInprogressRequest = true;
+    this.setHdsOptionItem()
+      .then(() => this.getSummary(this.model?.cart?.cartId))
+      .then((summary) => {
+        this.summary = summary;
```

Comment:
Why not directly using `getSummary` object from your resolve section ?

Reply:
This call is done in the context of the function `onHdsCheckboxChanged()`.
Another call to the API must be done to the update the contract data of the user.